x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -135,21 +154,24 @@ export function ServiceOverviewTransactionsTable({ serviceName }: Props) {
               transactionType,
               latencyAggregationType,
               transactionNames,
+              comparisonStart,
+              comparisonEnd,
             },
           },
         });
       }
     },
-    // only fetches statistics when requestId changes or transaction names changes
+    // only fetches statistics when requestId, transaction names or comparison range change
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [requestId, transactionNames],
+    [requestId, transactionNames, comparisonStart, comparisonEnd],
```

Comment:
What happens when comparisons are enabled, and the user selects a different date range? Will it cause the comparison request to happen twice?

Reply:
yes it will 🤬, I'll look into it.